api/v1beta1/cluster_types.go

```diff
@@ -514,6 +514,10 @@ type ClusterSpec struct {
 	// Version defines the temporal version the cluster to be deployed.
 	// This version impacts the underlying persistence schemas versions.
 	Version string `json:"version"`
+	// JobTtlSecondsAfterFinished is amount of time to keep job pods after jobs are completed.
+	// This field is immutable.
+	//+kubebuilder:validation:Minimum=1
+	JobTtlSecondsAfterFinished int32 `json:"jobTtlSecondsAfterFinished"`
```

Comment:
```suggestion
	JobTtlSecondsAfterFinished *int32 `json:"jobTtlSecondsAfterFinished"`
```

I think it would be easier to use it as a pointer to int32. I will allow the operator to make the distinction between value not set and `0` set as a value.